### PR TITLE
Pbi 8 6 determine and add spring dependencies

### PR DESCRIPTION
**Pull request by KingOnyxx**
Part of MOT-113
This pull request introduces a new `DependencyElements` class for generating Spring Boot project configuration files and adds corresponding unit tests. The most important changes include implementing the `DependencyElements` class, creating a Jinja2 template for `build.gradle.kts`, and adding test cases to validate the new functionality.

### New Feature: Spring Boot Dependency Management

* **`DependencyElements` class added**: A new class `DependencyElements` was created in `app/models/elements.py` to generate Spring Boot project configuration files using a Jinja2 template. It includes a method `print_django_style` (which should be renamed to `print_springboot_style`) for rendering the template with project-specific data.

* **Jinja2 template for `build.gradle.kts`**: Added a new template file `app/templates/springboot/build.gradle.kts.j2` to define the structure of the `build.gradle.kts` file. It dynamically incorporates dependencies, repositories, and project-specific settings.

### Testing Enhancements

* **Unit tests for `DependencyElements`**: Added a new test class `TestDependencyElements` in `tests/test_elements.py` to validate the functionality of the `DependencyElements` class. It includes positive, negative, and edge case tests for the `print_django_style` method.

* **Import update**: Updated imports in `tests/test_elements.py` to include the new `DependencyElements` class.

### CODE REVIEW

File: app/models/elements.py [hunk 1]
```diff
@@ -265,3 +265,43 @@ def print_django_style(self) -> str:
...
+
+
+class DependencyElements(FileElements):
+    """
+    This class is only for writing script for dependency in springboot framework
+    """
+
+    def print_django_style(  # change to print_springboot_style
+        self, project_name: str
+    ) -> str:
```
[the remark this answers is not in the document]
Comment (KenKomKom) [reply app/models/elements.py:277]: Rather than doing this, I suggest to rename the method "print_django_style" to something more generic like "print_with_style" so other file elements for future projects can also use itu

**Review by DaWanAnOnli — CHANGES_REQUESTED** [review]

File: app/models/elements.py [hunk 2]
```diff
@@ -265,3 +265,44 @@ def print_django_style(self) -> str:
...
         return bat
+
+
+class DependencyElements(FileElements):
+    """
+    This class is only for writing script for dependency in springboot framework
+    """
+
+    def print_django_style(self) -> str:
+        return super().print_django_style()
```
Comment (NtapSlur) [inline app/models/elements.py:276]: If the print_django_style() only calls the print_django_style() from the super(), it is fine not to write the def print_django_style for the DependencyElements since it will automatically inherit the method for print_django_style for the parent. For the rest, it is already good! Thankyou and keep up the good work!
